### PR TITLE
WIP : lots of dataraces, lots of mutexes

### PR DESCRIPTION
**Pull request by steveoc64**
Simple mutex around the []texts fixes some of the data races in the text widget.

However - there is still some horrible happening with the []objects that gets reset during refresh ???   not sure this is such a good plan, because that value is returned as part of the Objects() method ... the return value of which is used all over the place.

Having a think about that ...

### CODE REVIEW

File: widget/text.go [hunk 1]
```diff
@@ -181,24 +205,26 @@ func (t *textProvider) rowLength(row int) int {
...
 type textRenderer struct {
-	objects []fyne.CanvasObject
-
-	texts []*canvas.Text
-
-	provider *textProvider
+	objects    []fyne.CanvasObject
+	texts      []*canvas.Text
+	provider   *textProvider
+	textsMutex sync.RWMutex
```
Comment (paulhovey) [inline widget/text.go]: Is it safe to go with two mutexes for this?  Should there just be a single mutex that locks the whole textRenderer when in use?

File: widget/widget.go [hunk 2]
```diff
@@ -10,33 +10,42 @@ import (
...
 func (w *baseWidget) Size() fyne.Size {
+	//w.RLock()
+	//defer w.RUnlock()
 	return w.size
 }
 
 // Set a new size for a widget.
 // Note this should not be used if the widget is being managed by a Layout within a Container.
 func (w *baseWidget) resize(size fyne.Size, parent fyne.Widget) {
+	w.Lock()
```
Comment (paulhovey) [inline widget/widget.go]: I didn't realize you could just inherit properties like this, but makes sense since everything is a struct.  Cool!

**Comment by paulhovey** [comment]
Similar to https://github.com/fyne-io/fyne/pull/179

**Comment by steveoc64** [comment]
closed - experiment has suited its purpose

**Pull request closed without merging**